### PR TITLE
feature/17240-orgchart-rtl-hanging-lines

### PR DESCRIPTION
**Pull request by Xillez**
Added new option [organization.hangingSide](https://api.highcharts.com/highcharts/series.organization.hangingSide) which allows nodes to hang from the right side in inverted organization charts. See  #17240.

---

Asana task: https://app.asana.com/0/1204848917817812/1206243178116085/f

Also, added tests for both:
- Hanging lines (Shrink + Left side, Cumulative + Left side, Shrink + Right side, Cumulative + Right side).
- Bug where nodes don't drop in non-inverted organization chart (both "shrink" and "cumulative").

However, I wanted to mention this seems to add another issue with nodes being cut off by the plot area:
https://github.com/highcharts/highcharts/assets/5503724/68da3dea-3816-4e14-938c-6f12d29a7128
I'm unsure of whether this is within the scope of the task to fix? 
It can be avoided by switching to "shrink.
If it is inside the scope, got any good suggestions for solving this? maybe add shrink age to only those nodes that are clipping the edge?

### CODE REVIEW

**Comment by bre1470** [comment]
For me one of the unit tests is failing in Firefox because of a slightly different SVG path than expected.

Regarding the similar issue, it might be better to have this in a separate PR, just to be safe in case it evolves into something different of a solution.

**Comment by Xillez** [comment]
I'll check out the unit test!

The fix is very very simple, but for sure, I can separate them 👍

**Comment by Xillez** [comment]
I did a test and found that the node [clipping issue](https://github.com/highcharts/highcharts/assets/5503724/68da3dea-3816-4e14-938c-6f12d29a7128) mentioned is also present in master branch. I'll make an issue so it can be fixed separately if that is fine.

**Review by TorsteinHonsi — CHANGES_REQUESTED** [review]
Thanks!

The code looks great, nothing to add there.

* When it comes to the name of the new option, I think `hangingRight` as a boolean is a bit out of place. Usually when it comes to alignment we have enum-like string options like `align: 'left'|'center'|'right'`. So in this case I think we should rename (and refactor) the option to `hangingSide: 'left'|'right'`, with default value `left`. 
* Sample. We should have a visual sample, linked from the API doclet. Maybe using a real rtl is a bit hard for us, but a simple way to do it would be just to copy the `hanging-shrink` demo and do some minor changes to it to display the new feature. It will double as an educational sample and a visual test.

File: samples/highcharts/series-organization/hanging-right/demo.css
Comment (TorsteinHonsi) [inline samples/highcharts/series-organization/hanging-right/demo.css:1]: The sample should also be renamed to `hanging-side`

File: ts/Series/Organization/OrganizationSeriesDefaults.ts [hunk 1]
```diff
@@ -313,6 +313,19 @@ const OrganizationSeriesDefaults: OrganizationSeriesOptions = {
      * @private
      */
     hangingIndentTranslation: 'inherit',
+    /**
+     * Whether links connecting hanging nodes should be drawn on the left
+     * or right side. Useful for RTL layouts.
+     * **Note:** Only effects inverted charts (vertical layout).
+     *
+     * @sample highcharts/series-organization/hanging-right
```
Comment (TorsteinHonsi) [inline ts/Series/Organization/OrganizationSeriesDefaults.ts:321]: Remember to change the @sample link here too

**Review by TorsteinHonsi — CHANGES_REQUESTED** [review]
Thanks! Just a minor change now and it's a green light from me 👍

**Pull request merged**